src/Adapter/Address/CommandHandler/EditCustomerAddressHandler.php

```diff
@@ -57,12 +60,32 @@ public function handle(EditCustomerAddressCommand $command): void
                 throw new AddressConstraintException('Address contains invalid field values');
             }
 
-            if (false === $address->update()) {
+            // The address is used by an order so it is not edited directly, instead a copy is created and
+            if ($address->isUsed()) {
+                // Get a copy of current address
+                $old_address = new Address($address->id);
+
+                // Reset ID to force recreating a new address
+                $address->id = $address->id_address = null;
```

Comment:
I don't understand why we must reset the current address id and not the copied one? :thinking:

Reply:
Here the $address object is the new one, so its id must be null so that a new address is created. 

$old_address is what you call the "copied one", this one must keep its id.

Reply:
Exactly `$address` contains the edited data so this is the one that needs to be inserted in the DB, setting null to the id is kind of a trick but it makes the job